scala_config.bzl

```diff
@@ -0,0 +1,36 @@
+load("@bazel_skylib//lib:versions.bzl", "versions")
+load("//scala:scala_cross_version.bzl", "extract_major_version")
+
+def _default_scala_version():
+    """return the scala version for use in maven coordinates"""
+    return "2.12.11"
+
+def _store_config(repository_ctx):
+    bazel_version = versions.get()
+    repository_ctx.file("BUILD", "exports_files(['def.bzl'])")
+
+    scala_version = repository_ctx.attr.scala_version
+    scala_major_version = extract_major_version(scala_version)
+
+    config_file_content = "\n".join([
+        "BAZEL_VERSION='" + bazel_version + "'",
+        "SCALA_VERSION='" + scala_version + "'",
+        "SCALA_MAJOR_VERSION='" + scala_major_version + "'",
+    ])
+
+    repository_ctx.file("config.bzl", config_file_content)
+
+_config_repository = repository_rule(
+    implementation = _store_config,
+    attrs = {
+        "scala_version": attr.string(
+            mandatory = True,
+        ),
+    },
+)
+
+def scala_config(scala_version = _default_scala_version()):
```

Comment:
I'm not sure about `scala_config` naming. It contains only versions so maybe `versions` would be more appropriate. But as I understand it is a placeholder for other things as well. Maybe just `config` as it contains non scala related thing like `BAZEL_VERSION`? Either way I'm fine as it is currently. So you can just ignore this comment.

Reply:
I have intentionally named it config to allow other configuration to be stored in the future.